src/connection.rs

```diff
@@ -622,3 +621,243 @@ pub fn transaction<K: ToRedisArgs,
         }
     }
 }
+
+pub mod async {
```

Comment:
The file is already quite long and here we have another module.
IMO this should be extracted into its own file here.

Reply:
Just wanted to say. Also makes reading the diff hard because the types have the same name :)

Reply:
I could change to `AsyncConnection` etc but I feel it is more idiomatic to write `async::Connection` to avoid ambiguities in that case.